targets/TARGET_STM/TARGET_STM32F1/device/stm32f1xx_hal_smartcard.c

```diff
@@ -530,8 +529,7 @@ HAL_StatusTypeDef HAL_SMARTCARD_Receive(SMARTCARD_HandleTypeDef *hsc, uint8_t *p
       {
         return HAL_TIMEOUT;
       }
-      tmp = (uint16_t*) pData;
-      *tmp = (uint8_t)(hsc->Instance->DR & (uint8_t)0xFF);
+      *pData = (uint8_t)(hsc->Instance->DR & (uint8_t)0xFF);
```

Comment:
this is changing hal directly, will this be overwritten by next hal update?

cc @LMESTM @jeromecoutant

Reply:
Internal ticket reference : 39116

Reply:
> Internal ticket reference : 39116

What does this mean, that will be fixed in the next update. What about this change, +1 from your side?

Reply:
This means that this part of change is being reviewed internally.  But from review the change makes sense so I think we can accept this change in a first step and we will update it later if needed.

Reply:
Hi
Just received confirmation from driver team that patch in stm32f1xx_hal_smartcard.c is needed.
Thx for issue reporting